src/js/tabs/power.js

```diff
@@ -298,6 +300,32 @@ TABS.power.initialize = function (callback) {
             BATTERY_CONFIG.vbatwarningcellvoltage = parseFloat($('input[name="warningcellvoltage"]').val());
             BATTERY_CONFIG.capacity = parseInt($('input[name="capacity"]').val());
 
+            //scale calibrations
+            if (BATTERY_CONFIG.voltageMeterSource == 1) {
+                var vbatcalibration = parseFloat($('input[name="vbatcalibration-0"]').val());
+                if (vbatcalibration != 0) {
+                    var vbatnewscale = Math.round(VOLTAGE_METER_CONFIGS[0].vbatscale * (vbatcalibration / VOLTAGE_METERS[0].voltage));
+                    if (vbatnewscale >= 10 && vbatnewscale <= 255) {
+                        VOLTAGE_METER_CONFIGS[0].vbatscale = vbatnewscale;
+                    }
+                }
+            }
+            var ampsource = BATTERY_CONFIG.currentMeterSource;
+            if (ampsource == 1 || ampsource == 2) {
+                var amperagecalibration = parseFloat($('input[name="amperagecalibration-' + (ampsource - 1) + '"]').val());
+                if (amperagecalibration != 0) {
+                    var amperageoffset = CURRENT_METER_CONFIGS[ampsource - 1].offset / 1000;
+                    if (CURRENT_METERS[ampsource - 1].amperage == amperageoffset) {
+                        amperageoffset += 0.01;
```

Comment:
Why is this done?

Reply:
When scale is high enough, actual amperage will be same as offset, so in the actual calculation that would always cause 0 to be the result.

Reply:
I do not quite get how the scale being high should play into this - if the measured amperage is equal to the offset (i.e. the offset is wrong, or consumption is 0), then obviously it is not possible to calibrate. Instead of just fudging something in this case you should just not calibrate.

Reply:
I will work on the modal dialog with all the settings there and update this part as you said, just won't calibrate in that case.